src/parquet/column_reader.cc

```diff
@@ -186,6 +195,17 @@ std::shared_ptr<Page> SerializedPageReader::NextPage() {
       ParquetException::EofException(ss.str());
     }
 
+    std::vector<uint8_t> decrypt_buffer;
```

Comment:
Can you use a global decryption buffer? Please also allocate from the MemoryPool instead of allocating in the regular heap